python/lsst/daf/persistence/butlerLocation.py

```diff
@@ -93,7 +93,42 @@ class ButlerLocation(yaml.YAMLObject):
     persist and retrieve an object using the LSST Persistence Framework.
 
     Mappers should create and return ButlerLocations from their
-    map_{datasetType} methods."""
+    map_{datasetType} methods.
+
+    Parameters
+    ----------
+    pythonType - string or class instance
+        This is the type of python object that should be created when reading the location.
+
+    cppType - string or None
+        The type of cpp object represented by the location (optional, may be None)
+
+    storageName - string
+        The type of storage the object is in or should be place into.
+
+    locationList - list of string
```

Comment:
With the new component infrastructure, it seems to me that the previous uses of multiple entries in locationList are now obsolete and it could be renamed locationURI.